py-polars/tests/unit/operations/test_pivot.py

```diff
@@ -159,14 +159,31 @@ def test_pivot_multiple_values_column_names_5116() -> None:
     )
     expected = {
         "c1": ["A", "B"],
-        "x1|C": [1, 2],
-        "x1|D": [3, 4],
-        "x2|C": [8, 7],
-        "x2|D": [6, 5],
+        "x1|c2|C": [1, 2],
+        "x1|c2|D": [3, 4],
+        "x2|c2|C": [8, 7],
+        "x2|c2|D": [6, 5],
     }
     assert result.to_dict(False) == expected
 
 
+def test_pivot_duplicate_names_7731() -> None:
+    df = pl.DataFrame(
+        {"a": [1, 4], "b": [1, 2], "c": ["x", "x"], "d": [7, 8], "e": ["x", "y"]}
+    )
+    assert df.pivot(
+        values=["a", "d"], index="b", columns=["c", "e"], aggregate_function="first"
+    ).to_dict(False) == {
+        "b": [1, 2],
+        "a_c_x": [1, 4],
+        "d_c_x": [7, 8],
+        "a_e_x": [1, None],
+        "a_e_y": [None, 4],
+        "d_e_x": [7, None],
+        "d_e_y": [None, 8],
+    }
```

Comment:
did you have any thoughts on [the other suggestion](https://github.com/pola-rs/polars/issues/7731#issuecomment-1487237635), i.e. to have a column for each value and for each combination for the column values? currently this can be achieved with:
```python
In [28]: df.with_columns(c_e=pl.col('c').cast(pl.Utf8)+'_'+pl.col('e').cast(pl.Utf8)).pivot(values=["a", "d"], index="b", columns='c_e')
Out[28]:
shape: (2, 5)
┌─────┬───────┬───────┬───────┬───────┐
│ b   ┆ a_x_x ┆ a_x_y ┆ d_x_x ┆ d_x_y │
│ --- ┆ ---   ┆ ---   ┆ ---   ┆ ---   │
│ i64 ┆ i64   ┆ i64   ┆ i64   ┆ i64   │
╞═════╪═══════╪═══════╪═══════╪═══════╡
│ 1   ┆ 1     ┆ null  ┆ 7     ┆ null  │
│ 2   ┆ null  ┆ 4     ┆ null  ┆ 8     │
└─────┴───────┴───────┴───────┴───────┘
```

this would match pandas if the multiindex columns were squashed together:
```python
In [29]: df.to_pandas().pivot(values=["a", "d"], index="b", columns=['c', 'e'])
Out[29]:
     a         d
c    x         x
e    x    y    x    y
b
1  1.0  NaN  7.0  NaN
2  NaN  4.0  NaN  8.0
```
(it just seems a bit more complicated to implement, so I didn't think I could get it done by Friday)

Reply:
Nope, not yet. It is a nice to have indeed, but I will see if I can get it to work this release. Otherwise we do it one later. :+1:

Reply:
sure, I was just thinking that it might be more user-friendly to only make one breaking change rather than two, so if the `b   ┆ a_x_x ┆ a_x_y ┆ d_x_x ┆ d_x_y` solution was the one you wanted to adopt eventually, it might be better to wait and just do that one directly?

Reply:
The duplicate names should be fixed IMO. As that is a state a polars dataframe should not be in. So I will merge this one, and maybe I find time for the other change later this week. :)